src/main/scala/terminal/FileSystem.scala

```diff
@@ -0,0 +1,113 @@
+package terminal
+
+import java.util.NoSuchElementException
+
+
+class FileSystem {
+  //key = command;  value = how many paths need to be specified
+  val commandsAndNoOfPaths = Map("pwd" -> 0, "mkdir" -> 1, "ls" -> 0, "h" -> 0, "q" -> 0 )
+
+  var rootPath : Folders = _
+  var currentLevel : Folders = _
+  var cmdPrefix : String = ""
+  var path1_from : String = ""
+  var path2_to : String = ""
+
+
+
+  def run() : Unit = {
+    //initialize the root
+    rootPath = new Folders("")
+    rootPath.currentPath = "/"
+    currentLevel = rootPath
+
+    //run the terminal continuously until is chosen q to quit
+    commandInterpretor()
+  }
+
+
+  def commandInterpretor() : Unit = {
+    print( """Your cmd (press "q" to quit, "h" for help): """ )
+    val keyboardInput = getKeyboardInput
+
+    splitToCmdAndPaths( keyboardInput )
```

Comment:
How about
```suggestion
    splitToCmdAndPaths(getKeyboardInput).map(x => {
      splitToCmdAndPaths(x) match {
        case ....
      }
    }
```

Reply:
I ment 
```suggestion
    splitToCmdAndPaths(getKeyboardInput).map(x =>
    // propagate the input to the cmdPrefix
     cmdPrefix .. match {
       case "pwd"
       case "mkdir"
       ...
    }
```

Reply:
I like the idea, thanks, I will try to implement it.